CS3219-AY2324S1/ay2324s1-course-assessment-g56
# Add middleware that check for CORs headers and testcases for question, user and matching service

Adds testcases for question, user, matching service
Add middleware that check for CORs headers

Test cases for Matching Service(through a mock middleware that does not perform authorization checks):
- Queue 2 users that match difficulty
- Queue 1 user and timeout
- Queue 2 user with different difficulty and timeout
- Queue user but user initiate disconnect after 10 seconds
- Queue user but user stop finding pair after 10 seconds
- Queue the same user twice. User rejoin queue 1  second after failing match for initial queueing

Test cases for Question Service:
- CRUD operations on firebase endpoint(Checking of duplicates for create and update)
- Check whether CRUD can be performed if headers do not conform to those allowed under CORS 

Test cases for User Service:
- Check whether get requests can be processed if origin headers do not match the allowed headers through a mockApp that does not perform authorization checks for endpoints
